test/sqlancer/reducer/TestASTBasedReducer.java

```diff
@@ -0,0 +1,129 @@
+package sqlancer.reducer;
+
+import org.junit.jupiter.api.Test;
+import sqlancer.common.query.Query;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class TestASTBasedReducer {
+    @Test
+    void testLongStatement() throws Exception {
+        TestEnvironment env = TestEnvironment.getASTBasedReducerEnv();
+
+        String[] queriesStr = {
+                "SELECT DISTINCT * FROM v0 WHERE ((v0.rowid || ( (v0.c + v0.d) < 200 && v0.c >= 100) || 114514)OR(((v0.c0)||(1529686005)))) UNION SELECT DISTINCT * FROM v0 WHERE (NOT ((v0.rowid)OR(((v0.c0)||(1529686005))))) UNION SELECT DISTINCT * FROM v0 WHERE ((((v0.rowid)OR(((v0.c0)||(1529686005))))) IS NULL)" };
+        env.setInitialStatementsFromStrings(List.of(queriesStr));
+        env.setBugInducingCondition(statements -> {
+            String queriesString = TestEnvironment.getQueriesString(statements);
+            return queriesString.contains("&&");
+        });
+        env.runReduce();
+        List<Query<?>> reducedResult = env.getReducedStatements();
+        System.out.println(Arrays.toString(queriesStr));
```

Comment:
I guess we could already check against the result that this currently produces? We can still update the test case later when your implementation changes.